pkg/streamingpromql/operators/aggregation.go

```diff
@@ -152,61 +156,99 @@ func (a *Aggregation) NextSeries(ctx context.Context) (types.InstantVectorSeries
 		thisSeriesGroup := a.remainingInnerSeriesToGroup[0]
```

Comment:
`NextSeries()` is getting quite long - what do you think about breaking it into two methods? One for iterating through input series until the next group is complete, and another to compute the output series for the group

Reply:
Yes, but I think that will make reviewing this change more difficult. Given that the histogram changes slot in nicely, lets keep this as is and split apart in a new PR.